src/runtime_src/core/edge/user/shim.cpp

```diff
@@ -1545,12 +1545,19 @@ xclProbe()
   version.date_len = 128;
 
   int result = ioctl(fd, DRM_IOCTL_VERSION, &version);
-  if (result)
+  if (result) {
+    close(fd);
+    delete []version.name;
+    delete []version.desc;
+    delete []version.date;
     return 0;
+  }
 
   result = std::strncmp(version.name, "zocl", 4);
   close(fd);
-
+  delete []version.name;
+  delete []version.desc;
+  delete []version.date;
   return (result == 0) ? 1 : 0;
```

Comment:
Can we use std::string and populate the drm_version struct with name.c_str(), version.c_str() ?
The std::string can be constructed and initialized with `n` number of characters, e.g. `std::string name{128,0}`;
Raw new and delete should be avoided for exception safety reasons and to avoid memory leaks.

Reply:
done